CoreScriptsRoot/CoreScripts/Topbar.lua

```diff
@@ -957,17 +979,34 @@ local function Create3DSettingsIcon(topBarInstance, panel, menubar)
 
 		BackgroundTransparency = 1,
 
-		Image = "rbxasset://textures/ui/Menu/hamburger3D.png"
+		Image = "rbxasset://textures/ui/VR/hamburger.png"
 	}
 
+	function menuItem:OnSetActive(active)
+		MenuModule:SetVisibility(active)
+		if not active and MenuModule:GetVisibility() then
+			GuiService.MenuClosed:wait()
```

Comment:
We could either yield here or add a version of MenuModule:SetVisibility() that yields. Bad things happen if you don't wait for the animation to complete. Hm.